operator/pkg/util/apiclient/idempotency.go

```diff
@@ -230,6 +231,31 @@ func CreateOrUpdateStatefulSet(client clientset.Interface, statefuleSet *appsv1.
 	return nil
 }
 
+// CreateOrUpdateClusterRole creates a Clusterrole if the target resource doesn't exist. If the resource exists already, this function will update the resource instead.
+func CreateOrUpdateClusterRole(client clientset.Interface, clusterrole *rbacv1.ClusterRole) error {
+	_, err := client.RbacV1().ClusterRoles().Create(context.TODO(), clusterrole, metav1.CreateOptions{})
+
+	if err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			return err
+		}
+
+		older, err := client.RbacV1().ClusterRoles().Get(context.TODO(), clusterrole.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		clusterrole.ResourceVersion = older.ResourceVersion
+		_, err = client.RbacV1().ClusterRoles().Update(context.TODO(), clusterrole, metav1.UpdateOptions{})
```

Comment:
Maybe not need to update if it is created from operate ?

Reply:
i think when karamda instance need be updated, it will be useful, eg. add a new crd, then the clusterrole will be need update

Reply:
At the very least, the update operation will not lead to worse outcomes.